hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -269,7 +273,8 @@ public StorageStatistics provide() {
       }
       useListV1 = (listVersion == 1);
 
-      initTransferManager();
+      this.uploads = createUploadTransferManager(conf);
+      this.copies = createCopyTransferManager(conf);
```

Comment:
I'd actually like this to be on demand. That way, if not needed, no overhead of creating it. init times of s3a are already high

Reply:
Added a separate class called `LazyTransferManager` that lazily initializes a `TransferManager`.